core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/UnwrapYearInComparison.java

```diff
@@ -124,6 +129,36 @@ public Expression rewriteComparisonExpression(ComparisonExpression node, Void co
             return unwrapYear(expression);
         }
 
+        @Override
+        public Expression rewriteInPredicate(InPredicate node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            Expression inPredicate = treeRewriter.defaultRewrite((Expression) node, null);
+            Expression value = node.getValue();
+            Expression valueList = node.getValueList();
+
+            if (!(value instanceof FunctionCall call) ||
+                    !extractFunctionName(call.getName()).equals("year") ||
+                    call.getArguments().size() != 1 ||
+                    !(valueList instanceof InListExpression inListExpression)) {
+                return inPredicate;
+            }
+
+            // Convert each value to a comparison expression and try to unwrap it.
+            // unwrap the InPredicate only in case we manage to unwrap the entire value list
+            List<Expression> expressionsList = inListExpression.getValues();
+            ImmutableList.Builder<Expression> comparisonExpressionsListBuilder = ImmutableList.builder();
+            for (Expression currExpression : expressionsList) {
```

Comment:
- avoid abbreviations ("curr")
- `comparisonExpressionsListBuilder` is a long name and the fact it's listbuilder is known from its type
- `expressionsList` can be inlined


```suggestion
            ImmutableList.Builder<Expression> comparisonExpressions = ImmutableList.builderWithExpectedSize(inListExpression.getValues().size());
            for (Expression rightExpression : inListExpression.getValues()) {
                ComparisonExpression comparisonExpression = new ComparisonExpression(EQUAL, value, rightExpression);
```